### ISSUE TITLE
Word: Announce list item level?

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#6815]
Following on from issue #5267 and also correspondence with a user, is it possible to announce list level?

For instance, in Word, when you create a list, you can add items at "level 2" by pressing TAB when the focus is on a new list point.  Pressing TAB again moves to level 3 and so on.  Does Word disclose the list level and can we report this?

The complication is that everything can be customised - the amount of space before and after the bullet (or other symbol) at each level, so relying purely on formatting and spacing wouldn't be reliable.  Anyway, a question I thought was worth asking.  Visually, a sighted user isn't told that something is level 1 / level 2 etc, however they can see the surrounding text and tell how many levels there are in the document or this part of it at least.

**Comment by dkager** [comment]
> For instance, in Word, when you create a list, you can add items at "level 2" by pressing TAB when the focus is on a new list point. Pressing TAB again moves to level 3 and so on. Does Word disclose the list level and can we report this?

Yes, there is a patch already, though it needs some improvements: https://github.com/nvda-india/nvda/commit/91f26c716f101b66ab0820dd57e2f3718e62c543

There are quite a few points to consider regarding this feature.
This is an important feature for speech users because Word tends to use the same bullets for levels 1-4-7, 2-5-8 and 3-6-9. For braille this is also relevant because braille tables may not have distinct dot patterns for all possible bullets. So identifying list level based on bullet type is not always possible.

This is also relevant on the web. Granted, speech and braille announce the start and end of a list in this case, but sometimes it's useful to know the level too. Probably stuff for a separate issue, but good to keep in mind to keep things consistent.

One problem with the patch mentioned above is that it reports the level for each and every list item. Maybe that is not what speech users want, especially since the majority of documents will only have one or two list levels.
However, I wonder if this could be a good solution for braille. You could add "lv 1" to the end of a list item much as is already done for treeview items.
You could also rely on list start and list end announcements, but I don't know how well that works considering that in Word a list item can be an entire paragraph.
A different approach for braille users would be to indent sublevels by one or two spaces.

In summary, I feel this feature is important, but that the way the information is conveyed needs some careful thought.

**Comment by josephsl** [comment]
Hi,

Eight years later...

I doubt the linked pull request is going to work on NVDA 2025.2 code (alpha). Coupled with what Davy said about speech usage experiences, I think we need to thikn about the overall UX before working on a PR.

Thanks.
